### DISCUSSION TITLE
The following error originated from your test code, not from Cypress.  > process is not defined

### DISCUSSION THREAD

**Discussion by rajsudhirn23** [cypress-io/cypress#24535] [Questions and Help]
Hi, I'm facing the below error while running the unit test. Please can someone help to resolve this? I tried with Cypress.on('uncaught:exception', (err, runnable) => {return false }) on the path cypress\support\e2e.ts but didn't work :(

An uncaught error was detected outside of a test failed
No commands were issued in this test.
ReferenceError
The following error originated from your test code, not from Cypress.

> process is not defined

When Cypress detects uncaught errors originating from your test code it will automatically fail the current test.

Cypress could not associate this error to any specific test.

We dynamically generated a new test to display this failure.
[src/handlers/eoi/post_eoi.ts:30:13](http://localhost:57862/__/#)
  28 | 
  29 | const sqs = new SQS({
> 30 |     region: process.env.AWS_REGION,
     |             ^
  31 |     tls: true,
  32 | })
  33 |

**Comment by portyu9** [comment]
Hi @rajsudhirn23 

`process` is a Node.js global. Your Cypress test code is executing in the **browser**, so this:

```js
process.env.AWS_REGION
```

will throw `ReferenceError: process is not defined`.

`uncaught:exception` is not the right fix because this is an error in the test bundle/code itself, not an application exception that should be ignored.

If this SQS interaction uses AWS credentials, the preferred approach is to move it into Cypress's Node context using `cy.task()`:

```js
// cypress.config.js
setupNodeEvents(on, config) {
  on('task', {
    async doSqsWork(args) {
      const region = process.env.AWS_REGION

      // create/use the AWS client here

      return null
    },
  })
}
```

Then call it from the test:

```js
cy.task('doSqsWork', {
  // serializable arguments only
})
```

That also keeps AWS credentials out of the browser.

If `AWS_REGION` is merely non-sensitive public configuration and the SQS code genuinely needs to execute in the browser, use Cypress's browser configuration mechanism instead of `process.env`. On current Cypress, `Cypress.env()` has been removed: use `cy.env()` for sensitive values or `Cypress.expose()` for public synchronous configuration.

For AWS queue operations involving credentials, however, keeping the operation in `cy.task()` is the cleaner boundary.

Please mark this answer as accepted if it helped, thank you.
